### ISSUE TITLE
Double integrals problem

### ISSUE THREAD

**Issue by UlBoy** [sympy/sympy#21605]
#21533 
Hi guys. I want to solve a double integral formula with 'sympy' that is

I =  sy.Integral(exp(((0.236-1/z1)*(-0.488 + 0.5/z2 + 0.415/z1)+ (0.892-1/z1)*(-0.544 + 0.415/z2 + 0.5/z1)))*1/(z1**2) * 1/(z2**2), (z1,0,1),(z2,0,1))

I want to know the value calculated 'I' but it goes like this
![image](https://user-images.githubusercontent.com/84957780/121799256-02c30d80-cc66-11eb-96e1-8f4b3b144040.png)

which command do I have to use ? or Can't the expression itself be calculated?

Thanks for your helps

**Comment by oscarbenjamin** [comment]
You should just use a numerical integration routine. Ideally sympy would support evalf for double integrals but it does not yet. I already mentioned that in #21533. The difference with this example is just that it is not separable.

**Comment by UlBoy** [comment]
Thanks Oscar.
So are you means that there is no way to calculate for it right now ?
or should I separate formula in 2 single integrals ?

**Comment by UlBoy** [comment]
or should I try with 'mpmath'?

**Comment by oscarbenjamin** [comment]
This integral can be calculated using mpmath:
```python
In [7]: from mpmath import exp, quad

In [8]: f = lambda z1, z2: exp(((0.236-1/z1)*(-0.488 + 0.5/z2 + 0.415/z1)+ (0.892-1/z1)*(-0.544 + 0.415/z2 + 0.5/z1)))*1/(z1**2) * 1/(z2**2)

In [9]: quad(f, [0, 1], [0, 1])
Out[9]: mpf('0.51296909258032086')
```
SymPy should use mpmath to evaluate double integrals but currently this is only done for single integrals:
https://github.com/sympy/sympy/blob/47e86081be91ed386c7d27609c4aced55bdc3460/sympy/core/evalf.py#L1055-L1058

**Comment by UlBoy** [comment]
Thanks Oscar finally I got the answer with using mpmath !
I'm wondering how can I get the answer in 'PowerShell'
-----------------------------------------------------------------------------------------------------------
import sympy as sy
import mpmath as mp
sy.init_printing()
from sympy import *
from mpmath import exp
from mpmath import quad
from sympy import ln
from sympy import sqrt
from sympy import Matrix

x, u1, u2, z1, z2 = sy.symbols ("x, u1, u2, z1, z2")

x = 0.830 
V = Matrix ([[1,x],[x,1]])
det.V = V.det()**(-1/2) 


a = (2*3.14159)**-1 * det.V  # = 0.28534

u1 = ln(3.5/4.0)/sqrt(1.13**2+0.5**2) 
u2 = ln(0.81/1.0)/sqrt(0.19**2+0.2**2) 

f = lambda z1, z2: exp((((1+x)*((-1/z1)+u1+1)**2)+((1+x)*((-1/z2)+u2+1)**2))*(-1/2))*1/(z1**2)*1/(z2**2)
quad (f, [0,1],[0,1])

print (f)

P = a * f
print ('final answer is' P)
-----------------------------------------------------------------------------------------------------------------
PS C:\Users\6241\Documents\python\ssmrp> python sec.py
<function <lambda> at 0x000002302BDF4B80>
Traceback (most recent call last):
  File "sec.py", line 29, in <module>
    P = a * f
TypeError: unsupported operand type(s) for *: 'Float' and 'function'
------------------------------------------------------------------------------------------------------------------

I want to get 'P' which is multiple of 'a' and 'f' calculated with mpmath lambda and quad fucntion.
![image](https://user-images.githubusercontent.com/84957780/121845392-54c86980-cd20-11eb-8d24-89d5d810ccf0.png)
I did it with console but I can't get it .py file..
I'm sorry for disturbing you with basic skills.

**Comment by Joshua-Hwang** [comment]
Hi @oscarbenjamin, how difficult would you imagine implementing double/triple integrals using mpmath's `quad`?

I wouldn't mind trying my hand at this if you, or anyone else, is willing to review.

**Comment by oscarbenjamin** [comment]
Yes, go ahead. See the `evalf_integral` function in `sympy/core/evalf`. The single integral code can also be improved.

**Comment by Joshua-Hwang** [comment]
Done https://github.com/sympy/sympy/pull/25621

Let me know if you want me to change anything 😃.
